operator/internal/deploy/chart_manager.go

```diff
@@ -0,0 +1,222 @@
+
+type SkrChartConfig struct {
+	// WebhookChartPath represents the path of the webhook chart
+	// to be installed on SKR clusters upon reconciling kyma CRs.
+	WebhookChartPath             string
+	SkrWebhookMemoryLimits       string
+	SkrWebhookCPULimits          string
+	EnableWebhookPreInstallCheck bool
+}
+
+func NewEnabledSKRWebhookChartManager(config *SkrChartConfig) *EnabledSKRWebhookChartManager {
+	return &EnabledSKRWebhookChartManager{
+		config: config,
+	}
+}
+
+func (m *EnabledSKRWebhookChartManager) InstallWebhookChart(ctx context.Context, kyma *v1alpha1.Kyma,
+	remoteClientCache *remote.ClientCache, kcpClient client.Client,
+) (bool, error) {
+	skrClient, err := remote.NewRemoteClient(ctx, kcpClient, client.ObjectKeyFromObject(kyma),
+		kyma.Spec.Sync.Strategy, remoteClientCache)
+	if err != nil {
+		return true, err
+	}
+	skrCfg, err := remote.GetRemoteRestConfig(ctx, kcpClient, client.ObjectKeyFromObject(kyma),
+		kyma.Spec.Sync.Strategy)
+	if err != nil {
+		return true, err
+	}
+	argsVals, err := m.generateHelmChartArgs(ctx, kcpClient)
+	if err != nil {
+		return true, err
+	}
+	// TODO(khlifi411): make sure that validating-webhook-config resource is in sync with the secret configuration
+	skrWatcherInstallInfo := prepareInstallInfo(ctx, m.config.WebhookChartPath, ReleaseName, skrCfg, skrClient, argsVals)
+	err = installOrRemoveChartOnSKR(ctx, skrWatcherInstallInfo, ModeInstall, m.config.EnableWebhookPreInstallCheck)
+	if err != nil {
+		return true, err
+	}
+	kyma.UpdateCondition(v1alpha1.ConditionReasonSKRWebhookIsReady, metav1.ConditionTrue)
+	return false, nil
+}
+
+func (m *EnabledSKRWebhookChartManager) RemoveWebhookChart(ctx context.Context, kyma *v1alpha1.Kyma,
+	syncCtx *remote.KymaSynchronizationContext,
+) error {
+	skrCfg, err := remote.GetRemoteRestConfig(ctx, syncCtx.ControlPlaneClient, client.ObjectKeyFromObject(kyma),
+		kyma.Spec.Sync.Strategy)
+	if err != nil {
+		return err
+	}
+	argsVals, err := m.generateHelmChartArgs(ctx, syncCtx.ControlPlaneClient)
+	if err != nil {
+		return err
+	}
+	skrWatcherInstallInfo := prepareInstallInfo(ctx, m.config.WebhookChartPath, ReleaseName, skrCfg,
+		syncCtx.RuntimeClient, argsVals)
+	return installOrRemoveChartOnSKR(ctx, skrWatcherInstallInfo, ModeUninstall, m.config.EnableWebhookPreInstallCheck)
+}
+
+func (m *EnabledSKRWebhookChartManager) generateHelmChartArgs(ctx context.Context,
+	kcpClient client.Client,
+) (map[string]interface{}, error) {
+	watcherList := &v1alpha1.WatcherList{}
+	if err := kcpClient.List(ctx, watcherList, &client.ListOptions{}); err != nil {
+		return nil, fmt.Errorf("error listing watcher resources: %w", err)
+	}
+	if len(watcherList.Items) == 0 {
+		return nil, ErrFoundZeroWatchers
+	}
+	chartCfg := generateWatchableConfigs(watcherList)
+	bytes, err := k8syaml.Marshal(chartCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	kcpAddr, err := m.resolveKcpAddr(ctx, kcpClient)
+	if err != nil {
+		return nil, err
+	}
+	return map[string]interface{}{
```

Comment:
This is because stringifiedManifest is not cached based on `SetFlags`, bug in ModuleManager. I think this is a known cache limitation @adityabhatia but we should still make the cache optional so that people can disable it if they notice bugs

Reply:
@adityabhatia FYI